projects/core/src/product/store/effects/product-search-by-code.effect.ts

```diff
@@ -0,0 +1,116 @@
+              action.payload.scope
+          ),
+          mergeMap((group) => {
+            const scope = group.key;
+
+            return group.pipe(
+              map(
+                (action: ProductActions.ProductSearchLoadByCode) =>
+                  action.payload
+              ),
+
+              // We are grouping all load actions that happen at the same time
+              // to optimize loading and pass them all in a single call to the Connector.
+              bufferDebounceTime(debounce, scheduler),
+
+              mergeMap((payloads: { code: string; scope: string }[]) => {
+                const codes = payloads.map((payload) => payload.code);
+
+                return this.productSearchConnector
+                  .searchByCodes(codes, scope)
+                  .pipe(
+                    switchMap(
+                      (
+                        searchResults
+                      ): ProductActions.ProductSearchLoadByCodeSuccess[] => {
+                        // We got all products in a single response from the Connector,
+                        // but we need to dispatch separate success actions for each product.
+                        return searchResults.products?.map(
+                          (product, index) =>
+                            new ProductActions.ProductSearchLoadByCodeSuccess({
+                              // We assume that the order of the products in the response
+                              // is the same as the order of the codes in the request! OCC implementation guarantees that.
+                              ...payloads[index],
+                              product,
+                            })
+                        );
+                      }
+                    ),
+                    catchError(
+                      (error): ProductActions.ProductSearchLoadByCodeFail[] => {
+                        // We got an error while trying to load many products from the Connector,
+                        // but we need to dispatch separate fail actions for each product.
+                        return payloads.map(
+                          (payload) =>
+                            new ProductActions.ProductSearchLoadByCodeFail({
+                              ...payload,
+                              error: normalizeHttpError(error, this.logger),
+                            })
+                        );
+                      }
+                    )
+                  );
+              })
+            );
+          }),
+          withdrawOn(this.contextChange$)
+        )
+  );
+
+  clearState$: Observable<ProductActions.ClearProductSearchByCodeState> =
+    createEffect(() =>
+      this.actions$.pipe(
+        ofType(AuthActions.LOGOUT, AuthActions.LOGIN),
+        map(() => new ProductActions.ClearProductSearchByCodeState())
+      )
+    );
+
+  constructor(
+    private actions$: Actions,
+    private productSearchConnector: ProductSearchConnector
+  ) {}
```

Comment:
`major:` Let's not combine DI provided by the constructor and `inject` function. To keep our coding guideline, let's use only the `inject` function to prevent breaking changes. See: https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection

```suggestion
  // at the top of the class
  private actions$ = inject(Actions);
  private productSearchConnector = inject(ProductSearchConnector);
```

Reply:
Done